src/main/java/com/cloudbees/jenkins/plugins/bitbucket/hooks/HookProcessor.java

```diff
@@ -100,8 +100,11 @@ protected void scmSourceReIndex(final String owner, final String repository) {
                     // Search for the correct SCM source
                     if (source instanceof BitbucketSCMSource scmSource
                             && scmSource.getRepoOwner().equalsIgnoreCase(owner)
-                            && scmSource.getRepository().equals(repository)) {
+                            && scmSource.getRepository().equals(repository)
+                            && (mirrorId == null || mirrorId.equalsIgnoreCase(scmSource.getMirrorId()))) {
```

Comment:
@nfalco79 I noticed that this condition was missing. This could cause indexing on a primary (non mirror) repository multibranch when a mirror is syncrhonized. Which per my understand is unexpected.